compiler/src/main/java/org/qbicc/graph/OrderedNode.java

```diff
@@ -1,26 +1,15 @@
 package org.qbicc.graph;
 
-import java.util.NoSuchElementException;
-
 /**
  * A node which may be ordered after another node in the program order.
  */
 public interface OrderedNode extends Node {
-    /**
-     * Determine whether this node has a program-ordered predecessor.
-     *
-     * @return {@code true} if this node has a program-ordered predecessor, {@code false} otherwise
-     */
-    default boolean hasDependency() {
```

Comment:
This was confusing. Glad it is removed and we have a separate node for "no-side-effects" calls.